runtime/test_artifacts/generator.go

```diff
@@ -6,108 +6,149 @@ import (
 		if value == int32(coreTypes.Pools_POOLS_UNSPECIFIED) {
 			continue
 		}
 
+		// TECHDEBT: Test artifact should reflect the sum of the initial account values
+		// rather than be set to `DefaultAccountAmountString`
 		amount := DefaultAccountAmountString
 		if value == int32(coreTypes.Pools_POOLS_FEE_COLLECTOR) {
-			amount = "0"
+			amount = "0" // fees are empty at genesis
 		}
 
+		poolAddr := hex.EncodeToString(coreTypes.Pools(value).Address())
+
 		pools = append(pools, &coreTypes.Account{
-			Address: hex.EncodeToString(coreTypes.Pools(value).Address()),
+			Address: poolAddr,
 			Amount:  amount,
 		})
 	}
-	return
+	return pools
+}
+
+func newAccountsWithKeys(privateKeys []string) (accounts []*coreTypes.Account) {
+	for _, pk := range privateKeys {
+		pk, _ := crypto.NewPrivateKey(pk)
+		addr := pk.Address().String()
+		accounts = append(accounts, &coreTypes.Account{
+			Address: addr,
+			Amount:  DefaultAccountAmountString,
+		})
+	}
+	return accounts
 }
 
-func NewAccounts(n int, privateKeys ...string) (accounts []*coreTypes.Account) {
-	for i := 0; i < n; i++ {
+//nolint:unused
+func newAccounts(numActors int) (accounts []*coreTypes.Account) {
+	for i := 0; i < numActors; i++ {
 		_, _, addr := keygen.GetInstance().Next()
-		if privateKeys != nil {
-			pk, _ := crypto.NewPrivateKey(privateKeys[i])
-			addr = pk.Address().String()
-		}
 		accounts = append(accounts, &coreTypes.Account{
 			Address: addr,
 			Amount:  DefaultAccountAmountString,
 		})
 	}
-	return
+	return accounts
 }
 
-// TODO: The current implementation of NewActors  will have overlapping `ServiceUrl` for different
-//
-//	types of actors which needs to be fixed.
-func NewActors(actorType coreTypes.ActorType, n int) (actors []*coreTypes.Actor, privateKeys []string) {
-	for i := 0; i < n; i++ {
+// TECHDEBT: Current implementation of `NewActors` will result in non-unique ServiceURLs if called
+// more than once.
+func NewActors(actorType coreTypes.ActorType, numActors int, chains []string) (actors []*coreTypes.Actor, privateKeys []string) {
+	// If the actor type is a validator, the chains must be nil since they are chain agnostic
+	if actorType == coreTypes.ActorType_ACTOR_TYPE_VAL {
+		logger.Global.Warn().Msgf("validator actors should not have chains but a list was provided: %v", chains)
+		chains = nil
+	}
+	for i := 0; i < numActors; i++ {
 		serviceURL := getServiceURL(i + 1)
-		actor, pk := NewDefaultActor(int32(actorType), serviceURL)
+		actor, pk := NewDefaultActor(actorType, serviceURL, chains)
 		actors = append(actors, actor)
 		privateKeys = append(privateKeys, pk)
 	}
-
-	return
+	return actors, privateKeys
 }
 
-func getServiceURL(n int) string {
-	return fmt.Sprintf(ServiceURLFormat, n)
-}
-
-func NewDefaultActor(actorType int32, serviceURL string) (actor *coreTypes.Actor, privateKey string) {
+func NewDefaultActor(actorType coreTypes.ActorType, serviceURL string, chains []string) (actor *coreTypes.Actor, privateKey string) {
```

Comment:
nit: multi-line?

Reply:
I love multilining method signatures so never hesitate to ask  :)